lib/image.py

```diff
@@ -435,18 +436,25 @@ def tarballs_write(self, tarball_dir):
       # it when (if) we have multiple layers. But, I wanted the interface to
       # support multiple layers.
       base = "%s.tar" % self.ref.for_path
+      base_hash = None
       path = tarball_dir // base
       try:
          ch.INFO("layer 1/1: gathering")
          ch.VERBOSE("writing tarball: %s" % path)
-         fp = fs.TarFile.open(path, "w", format=tarfile.PAX_FORMAT)
-         unpack_path = self.unpack_path.resolve()  # aliases use symlinks
-         ch.VERBOSE("canonicalized unpack path: %s" % unpack_path)
-         fp.add_(unpack_path, arcname=".")
-         fp.close()
+         bytes_ = None
+         with io.BytesIO() as buf:
```

Comment:
The idea was to calculate the uncompressed layer hash during the layer tarFile creation, rather than re-reading the file again later. TarFile is kinda hairy to me, unclear if I can read and write in chunks with it as the tar is being created.